userspace/libsinsp/filter/ast.cpp

```diff
@@ -19,6 +19,17 @@ limitations under the License.
 
 using namespace libsinsp::filter::ast;
 
+// Included in parser.h, not including parser.h here to avoid circular
```

Comment:
I see why you're doing this, but this can get very fragile. We also have many cases like this in sinsp and this definition-import pattern is getting harder and harder to manage.

At this point, I'd suggest having creating two standalone `escaping.h` and `escaping.cpp`. The `.h` would define:
```
namespace libsinsp {
namespace filter {
std::string escape(const std::string& str);
std::string unescape(const std::string& str);
};
};
```

Then, both AST and parser could import and depend on that. I think this would make this much more manageable and reusable across the codebase. What do you think?